code/datums/components/customizable_reagent_holder.dm

```diff
@@ -110,6 +110,8 @@
 /datum/component/customizable_reagent_holder/proc/valid_ingredient(obj/ingredient)
 	if (HAS_TRAIT(ingredient, TRAIT_CUSTOMIZABLE_REAGENT_HOLDER))
 		return FALSE
+	if(isgrenade(ingredient))
```

Comment:
Could you make this look for a trait instead? TRAIT_BURGERABLE_NON_FOOD or something so this doesn’t devolve into a list of istype checks whenever someone else wants to add make non-food object be addable to food